components/centraldashboard-angular/frontend/src/app/services/environment.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { DashboardLinks } from '../types/dashboard-links';
+import { envInfo } from '../types/env-info';
+import { Namespace } from '../types/namespace';
+import { PlatformInfo } from '../types/platform-info';
+import { CDBBackendService } from './backend.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EnvironmentService {
+  public platform = new ReplaySubject<PlatformInfo>();
+  public user = new ReplaySubject<string>();
+  public namespaces = new ReplaySubject<Namespace[]>();
+  public dashboardLinks = new ReplaySubject<DashboardLinks>();
+
+  constructor(private backend: CDBBackendService) {
+    this.backend.getEnvInfo().subscribe((res: envInfo) => {
+      if (res.user) {
+        this.user.next(res.user);
+      }
+      if (res.platform) {
+        this.platform.next(res.platform);
+      }
+      if (res.namespaces) {
+        this.namespaces.next(res.namespaces);
+      }
+    });
```

Comment:
Do we really need 3 different subjects for these values?

Reply:
We don't really need 3 different subjects since we fetch all of the above simultaneously from the same request. I followed the above approach because I found it to be cleaner since some components may need only one of these properties. However, I 'm open to reconsidering it, since this results in some other components having to subscribe more than once.

Reply:
I don't have a strong opinion either. So, let's not change it.